packages/controller-utils/src/types.ts

```diff
@@ -13,11 +13,22 @@ export const InfuraNetworkType = {
 export type InfuraNetworkType =
   (typeof InfuraNetworkType)[keyof typeof InfuraNetworkType];
 
+/**
+ * Custom network types that are not part of Infura.
+ */
+export type CustomNetworkType = 'megaeth-testnet';
```

Comment:
nit:

lets do the same as InfuraNetworkType to align the behaviour, and using that may allow you to enable mulitple custom networks later

```suggestion
export const CustomNetworkType = {
  megaeth-testnet: 'megaeth-testnet',
} as const;

export type CustomNetworkType =
  (typeof CustomNetworkType)[keyof typeof CustomNetworkType];
```